chainstate/src/detail/mod.rs

```diff
@@ -191,14 +197,17 @@ impl<S: BlockchainStorage, V: TransactionVerificationStrategy> Chainstate<S, V>
         time_getter: TimeGetter,
     ) -> Self {
         let orphan_blocks = OrphansProxy::new(*chainstate_config.max_orphan_blocks);
+        let events_controller = EventsController::new();
+        let event_broadcast = broadcaster::Broadcaster::new();
         Self {
             chain_config,
             chainstate_config,
             chainstate_storage,
             tx_verification_strategy,
             orphan_blocks,
             custom_orphan_error_hook,
-            events_controller: EventsController::new(),
+            events_controller,
+            event_broadcast,
```

Comment:
Here is the bad stuff. There are now two mechanisms for delivering and subscribing to events. The `EventController` does not seem to have an easy way to clean up dead subscriptions, producing a risk of memory exhaustion even in expected non-adversial scenarios. The implementation is also somewhat weird with the thread pool in place just to deliver events. On the other hand, the newly introduced `Broadcaster` lacks some features like the ability to wait until all events are processed.

Having both is highly unsatifying and error prone. There's also the potential issue with changing the event mechanism last minute and whatever unintended consequences that might cause in case we unify them in any way. Any suggestions?

Reply:
As discussed, this is OK, because these serve different purposes.